conan/tools/apple/apple.py

```diff
@@ -167,9 +167,14 @@ def _get_install_name(path_to_dylib):
         install_name = check_output_runner(command).strip().split(":")[1].strip()
         return install_name
 
-    def _darwin_collect_dylibs(lib_folder):
-        return [os.path.join(lib_folder, f) for f in os.listdir(lib_folder) if f.endswith(".dylib")
-                and not os.path.islink(os.path.join(lib_folder, f))]
+    def _darwin_is_binary(file, binary_type):
+        if binary_type not in ("DYLIB", "EXECUTE") or os.path.islink(file) or os.path.isdir(file):
+            return False
+        check_file = f"otool -hv {file}"
+        return binary_type in check_output_runner(check_file)
```

Comment:
I suppose `check_output_runner` may raise, therefore any error from `otool` invocation (like, it failed to parse a particular file, or just missing), will result in failure to create a package.
should it have `try/catch` block here to return `False` in case of any errors?